thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/diffsummary/MultiDimensionalSummary.java

```diff
@@ -0,0 +1,290 @@
+        .build();
+    options.addOption(hierarchies);
+
+    Option oneSideError = Option.builder("oneSideError").desc("enable one side error summary").build();
+    options.addOption(oneSideError);
+
+    Option manualOrder = Option.builder("manualOrder").desc("use manual dimension order").build();
+    options.addOption(manualOrder);
+
+    Option dateTimeZone =
+        Option.builder("timeZone").desc("time zone id in Joda library").hasArg().argName("ID").build();
+    options.addOption(dateTimeZone);
+
+    Option costFunctionClass = Option.builder("cost").longOpt("costFunction").desc(
+        "the parameters of the cost function (a map in json format) "
+            + "Essential field in the map: 'className'").hasArg().argName("JSON").build();
+    options.addOption(costFunctionClass);
+
+    return options;
+  }
+
+  /**
+   * Removes noisy dimensions.
+   *
+   * @param dimensions the original dimensions.
+   *
+   * @return the original dimensions minus noisy dimensions, which are predefined.
+   *
+   * TODO: Replace with an user configurable method
+   */
+  public static Dimensions sanitizeDimensions(Dimensions dimensions) {
+    List<String> allDimensionNames = dimensions.allDimensions();
+    Set<String> dimensionsToRemove = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
+    dimensionsToRemove.add("environment");
+    dimensionsToRemove.add("colo");
+    dimensionsToRemove.add("fabric");
+    for (String dimensionName : allDimensionNames) {
+      if(dimensionName.contains(TOP_K_POSTFIX)) {
+        String rawDimensionName = dimensionName.replaceAll(TOP_K_POSTFIX, "");
+        dimensionsToRemove.add(rawDimensionName.toLowerCase());
+      }
+    }
+    return removeDimensions(dimensions, dimensionsToRemove);
+  }
+
+  private static Dimensions removeDimensions(Dimensions dimensions, Collection<String> dimensionsToRemove) {
+    List<String> dimensionsToRetain = new ArrayList<>();
+    for (String dimensionName : dimensions.allDimensions()) {
+      if(!dimensionsToRemove.contains(dimensionName)){
+        dimensionsToRetain.add(dimensionName);
+      }
+    }
+    return new Dimensions(dimensionsToRetain);
+  }
+
+  private static CostFunction parseCostFunction(String paramString)
+      throws IOException, ClassNotFoundException, NoSuchMethodException, IllegalAccessException,
+      InvocationTargetException, InstantiationException {
+    HashMap<String, String> params = objectMapper.readValue(paramString, HashMap.class);
+
+    String className = params.get("className");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(className), "Class name of cost function cannot be empty.");
+
+    Class<CostFunction> clazz = (Class<CostFunction>) Class.forName(className);
+    Constructor<CostFunction> constructor = clazz.getConstructor(Map.class);
+    CostFunction costFunction = constructor.newInstance(new Object[] { params });
+
+    return costFunction;
+  }
+
+  public static void main(String[] args) throws Exception {
```

Comment:
Is this intended as a stand-alone CLI, or as a service?

Reply:
Yes, the stand-alone CLI is created for our users to conduct quick experiments without bringing up the entire ThirdEye.